include/envoy/init/init.h

```diff
@@ -32,8 +34,10 @@ class Manager {
   /**
    * Register a target to be initialized in the future. The manager will call initialize() on each
    * target at some point in the future. It is an error to register the same target more than once.
+   * @param target the Target to initialize.
+   * @param description a human-readable description of target used for logging and debugging.
    */
-  virtual void registerTarget(Target& target) PURE;
+  virtual void registerTarget(Target& target, absl::string_view description) PURE;
```

Comment:
I'm wondering if it'd be a bit cleaner to add a `description()` method to `Init::Target`, making them self-describing, rather than adding the parameter here and the `TargetWithDescription` alias below.

Reply:
It would be, but that would add more burden to the callers. I don't mind hiding the ugly in `InitManagerImpl`, but if other folks feel strongly, we can do this.

Reply:
Don't callers have this burden anyway, having to pass an additional param to `registerTarget`?

Reply:
Yeah, but they don't have to implement a method.